src/pages/workspaces/workspace/analysis/modals/AnalysisModal.ts

```diff
@@ -247,60 +286,59 @@ export const AnalysisModal = withDisplayName('AnalysisModal')(
-
       const errors = validate(
         { analysisName: `${analysisName}.${fileExt}`, notebookKernel },
         {
           analysisName: analysisNameValidator(_.map(({ name }) => getFileName(name), analyses)),
           notebookKernel: { presence: { allowEmpty: true } }
         }
       )
-
       return div({ style: { display: 'flex', flexDirection: 'column' } }, [
-        h(IdContainer, [id => h(Fragment, [
-          h(FormLabel, { htmlFor: id, required: true }, [`Name of the ${getArtifactLabel(toolLabel)}`]),
+        h(Fragment, [
+          h(FormLabel, { htmlFor: analysisNameInputId, required: true }, [`Name of the ${getArtifactLabel(toolLabel)}`]),
           analysisNameInput({
             error: Utils.summarizeErrors(prevAnalysisName !== analysisName && errors?.analysisName),
             inputProps: {
-              id, value: analysisName,
+              id: analysisNameInputId, value: analysisName,
               onChange: v => {
                 setAnalysisName(v)
               }
             }
           })
-        ])]),
-        Utils.cond(
-          [isJupyterLab || isJupyter, () => h(IdContainer,
-            [id => h(Fragment, [
-              h(FormLabel, { htmlFor: id, required: true }, ['Language']),
-              h(Select, {
-                id, isSearchable: true,
-                placeholder: 'Select a language',
-                getOptionLabel: ({ value }) => _.startCase(value),
-                value: notebookKernel,
-                onChange: ({ value: notebookKernel }) => setNotebookKernel(notebookKernel),
-                options: ['python3', 'r']
-              })
-            ])]
-          )],
-          [isRStudio, () => h(IdContainer, [id => h(Fragment, [
-            h(FormLabel, { htmlFor: id, required: true }, ['File Type']),
+        ]),
+        (isJupyterLab || isJupyter) &&
+          h(Fragment, [
+            h(FormLabel, { htmlFor: nameSelectId, required: true }, ['Language']),
             h(Select, {
-              id, isSearchable: true,
-              value: fileExt,
-              onChange: v => {
-                setFileExt(v.value)
-              },
-              options: toolExtensionDisplay.RStudio
+              id: nameSelectId, isSearchable: true,
+              placeholder: 'Select a language',
+              getOptionLabel: ({ value }) => _.startCase(value),
+              value: notebookKernel,
+              onChange: ({ value: notebookKernel }) => setNotebookKernel(notebookKernel),
+              options: ['python3', 'r']
             })
-          ])])]
-        ),
+          ]),
+        isRStudio && h(Fragment, [
+          h(FormLabel, { htmlFor: fileTypeSelect, required: true }, ['File Type']),
+          h(Select, {
+            id: fileTypeSelect, isSearchable: true,
+            value: fileExt,
+            onChange: v => {
+              setFileExt(v.value)
+            },
+            options: toolExtensionDisplay.RStudio as ExtensionDisplay[]
```

Comment:
Is this cast necessary? Isn't `toolExtensionDisplay.RStudio` already typed as `ExtensionDisplay[]`?

Reply:
It's unhappy about the possibility of being undefined.

What should I do in this situation?

Reply:
Ooh, since it's Partial (`Partial<Record<ToolLabel, ExtensionDisplay[]>>`). I'm not sure.

In that case, I think this could be written more concisely as `toolExtensionDisplay.RStudio!`.